### ISSUE TITLE
QQ.algebraic_field(Rational) should be just QQ

### ISSUE THREAD

**Issue by skirpichev** [sympy/sympy#14476]
or raise some exception.

Following doesn't have any sense:
```python
In [2]: QQ.algebraic_field(Rational(1, 7))
Out[2]: QQ<1/7>
```

**Comment by jksuom** [comment]
I think `QQ.algebraic_field(r)` for a rational `r` (including 0 and 1) or a sequence of rational numbers should be `QQ`. Raising an exception is probably not necessary.

**Comment by abaokar-07** [comment]
If the following change is made in `sympy/polys/numberfields.py`
```
if ex.is_Rational:
     raise SomeException("description")
```
then `QQ.algebraic_field(Rational(1, 7))` is also raising exception, but other examples are failing in good numbers

**Comment by skirpichev** [comment]
> I think QQ.algebraic_field(r) for a rational r (including 0 and 1) or a sequence of rational numbers should be QQ.

So, QQ in this context should be understood as degree=0 field extension of QQ itself?

> If the following change is made ... then

Then don't do this.  Proper fix in the way @jksuom suggested - should, probably, introduce ``__new__`` method for AlgebraicField.

**Comment by jksuom** [comment]
> degree=1 field extension of QQ itself?

That would be natural. `__new__` would probably be needed (unless RationalField could be made a subclass of AlgebraicField).

**Comment by abaokar-07** [comment]
@jksuom @skirpichev What should `__new__`  contain?

**Comment by skirpichev** [comment]
On Sun, Mar 11, 2018 at 03:13:30AM -0700, Kalevi Suominen wrote:
>    unless RationalField could be made a subclass of AlgebraicField).

Probably, in this case too.  But I'm not sure that subclassing
may have some mathematical sense in this case.

**Comment by jksuom** [comment]
> Probably, in this case too.

I agree. Though I think that RationalField could be a subclass of AlgebraicField (finite algebraic extension of degree 1) from the mathematical point of view. The situation is analogous to Integer being derived Rational (rational number with denominator 1).

**Comment by abaokar-07** [comment]
So subclassing will be enough in this case?

**Comment by jksuom** [comment]
No, I don't think that is enough. But at some later point it may become useful.
